src/xpk/core/cluster_private.py

```diff
@@ -0,0 +1,170 @@
+
+
+def authorize_private_cluster_access_if_necessary(args) -> int:
+  """Updates a GKE cluster to add authorize networks to access a private cluster's control plane, if not added already.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and error code otherwise.
+  """
+  if not is_cluster_private(args):
+    if not args.private and args.authorized_networks is None:
+      xpk_print('Cluster is public and no need to authorize networks.')
+      return 0
+    else:
+      xpk_print(
+          'Cannot convert an existing public cluster to private. The arguments'
+          ' --private and --authorized-networks are not acceptable for public'
+          ' clusters.'
+      )
+      return 1
+
+  new_authorized_networks_needed = args.authorized_networks is not None
+
+  authorized_networks = (
+      args.authorized_networks
+      if new_authorized_networks_needed
+      else get_cluster_authorized_networks(args)
+  )
+
+  if not new_authorized_networks_needed:
+    return_code, current_machine_already_authorized = (
+        is_current_machine_in_any_network(authorized_networks)
+    )
+    if return_code > 0:
+      xpk_print("Error checking current machine's IP address.")
+      return return_code
+    new_authorized_networks_needed = not current_machine_already_authorized
+
+  if new_authorized_networks_needed:
+    return_code, authorized_networks = add_current_machine_to_networks(
+        authorized_networks
+    )
+    if return_code > 0:
+      xpk_print(
+          "Adding current machine's IP address to the authorized networks"
+          ' failed!'
+      )
+      return return_code
+
+    cluster_authorized_networks_update_code = (
+        update_cluster_authorized_networks(args, authorized_networks)
+    )
+    if cluster_authorized_networks_update_code > 0:
+      xpk_print('Updating cluster authorized networks failed!')
+      return cluster_authorized_networks_update_code
+    xpk_print("Cluster's master authorized networks updated successfully.")
+  else:
+    xpk_print('Current machine is already in authorized networks.')
+
+  return 0
+
+
+def is_cluster_private(args) -> bool:
+  """Checks if cluster is private.
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    True if cluster is private and False otherwise.
+  """
+  command = (
+      f'gcloud container clusters describe {args.cluster}'
+      f' --project={args.project} --region={zone_to_region(args.zone)}'
+      ' --format="value(privateClusterConfig.enablePrivateNodes)"'
+  )
+  return_code, private_nodes_enabled = run_command_for_value(
+      command,
+      'Check if Private Nodes is enabled in cluster.',
+      args,
+  )
+
+  if return_code != 0:
+    xpk_print('Checking if Private Nodes is enabled failed!')
+    xpk_exit(return_code)
+
+  if private_nodes_enabled.strip().lower() == 'true':
```

Comment:
This function would make a good util, something like: is_text_true

Reply:
done